### PR TITLE
2.5.x

### PR DESCRIPTION
**Pull request by shenxinfeng88**
## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/alibaba/dubbo/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by chickenlj** [comment]
Please describe the purpose of this PR and the changes involved. And also this PR contains too many unrelated changes, which really confuses me.

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,9 @@ jdk:
     - oraclejdk8
     - openjdk7
 
-script: "mvn cobertura:cobertura"
+script:
+    - mvn clean package
```
Comment (kimmking) [inline .travis.yml]: no nessesory.

mvn cober... will package project first

File: dubbo-registry/dubbo-registry-api/src/main/java/com/alibaba/dubbo/registry/support/AbstractRegistry.java [hunk 2]
```diff
@@ -442,10 +439,6 @@ private void saveProperties(URL url) {
     }
 
     public void destroy() {
-        if (!destroyed.compareAndSet(false, true)) {
```
Comment (kimmking) [inline dubbo-registry/dubbo-registry-api/src/main/java/com/alibaba/dubbo/registry/support/AbstractRegistry.java]: why remove it?

File: hessian-lite/src/main/java/com/alibaba/com/caucho/hessian/io/AbstractDeserializer.java [hunk 3]
```diff
@@ -75,11 +75,27 @@ public Object readList(AbstractHessianInput in, int length)
         throw new UnsupportedOperationException(String.valueOf(this));
     }
 
+    @Override
+    public Object readList(AbstractHessianInput in, int length, Class<?> expectType) throws IOException {
```
Comment (kimmking) [inline hessian-lite/src/main/java/com/alibaba/com/caucho/hessian/io/AbstractDeserializer.java]: no nessesory.

File: hessian-lite/src/main/java/com/alibaba/com/caucho/hessian/io/AbstractHessianInput.java [hunk 4]
```diff
@@ -334,12 +335,34 @@ abstract public InputStream readInputStream()
...
+    /**
+     * Reads an arbitrary object from the input stream.
+     *
+     * @param expectedClass the expected class if the protocol doesn't supply it.
+     * @param expectedTypes the runtime type hints, eg: expectedClass equals Map, expectedTypes can
+     * equals String.class, Short.class
+     */
+    public Object readObject(Class expectedClass, Class<?>... expectedTypes)
+        throws IOException{
+        throw new UnsupportedOperationException(String.valueOf(this));
```
Comment (kimmking) [inline hessian-lite/src/main/java/com/alibaba/com/caucho/hessian/io/AbstractHessianInput.java]: why

File: hessian-lite/src/main/java/com/alibaba/com/caucho/hessian/io/Hessian2Input.java [hunk 5]
```diff
@@ -2051,7 +2054,12 @@ private HashMap readFault()
      */
     public Object readObject(Class cl)
             throws IOException {
-        if (cl == null || cl == Object.class)
+        return readObject(cl, null, null);
+    }
+
+    @Override
+    public Object readObject(Class expectedClass, Class<?>... expectedTypes) throws IOException {
+        if (expectedClass == null || expectedClass == Object.class)
```
Comment (kimmking) [inline hessian-lite/src/main/java/com/alibaba/com/caucho/hessian/io/Hessian2Input.java]: why move check code here?

Comment (zonghaishang) [reply hessian-lite/src/main/java/com/alibaba/com/caucho/hessian/io/Hessian2Input.java]: not remove code , just refactor code.

**Pull request merged**